bluesky/log.py

```diff
@@ -125,22 +136,200 @@ def format(self, record):
         return formatted.replace("\n", "\n    ")
 
 
-plain_log_format = "[%(levelname)1.1s %(asctime)s.%(msecs)03d %(module)s:%(lineno)d] %(message)s"
+plain_log_format = "[%(levelname)1.1s %(asctime)s.%(msecs)03d %(module)15s:%(lineno)5d] %(message)s"
 color_log_format = ("%(color)s[%(levelname)1.1s %(asctime)s.%(msecs)03d "
-                    "%(module)s:%(lineno)d]%(end_color)s %(message)s")
+                    "%(module)15s:%(lineno)5d]%(end_color)s %(message)s")
+
+
+def color_logs(color):
+    """
+    If True, add colorful logging handler and ensure plain one is removed.
+
+    If False, do the opposite.
+    """
+    warnings.warn(f"The function color_logs is deprecated. "
+                  f"Use `config_bluesky_logging(color={color})` instead.")
+    config_bluesky_logging(color=color)
+
+
 logger = logging.getLogger('bluesky')
+doc_logger = logging.getLogger('bluesky.emit_document')
+msg_logger = logging.getLogger('bluesky.msg')
+status_logger = logging.getLogger('bluesky.status')
+current_handler = None
+
+
+def validate_level(level) -> int:
+    '''
+    Return a int for level comparison
 
+    '''
+    if isinstance(level, int):
+        levelno = level
+    elif isinstance(level, str):
+        levelno = logging.getLevelName(level)
+
+    if isinstance(levelno, int):
+        return levelno
+    else:
+        raise ValueError("Your level is illegal, please use one of python logging string")
 
-current_handler = None  # overwritten below
 
+class DocFilter(logging.Filter):
+    '''
+    Block any message that is lower than certain level except it related to target
+    documents. You have option to choose whether env, config and misc message is exclusive
+    or not.
+
+    Parameters
+    ----------
+    names : string or list of string
+        Name of document type list which will be filtered in, one of 'Start', 'Descriptor', 'Event' and 'Stop'.
```

Comment:
The strings should be lowercase, as in the convention everywhere else in bluesky. And datum and resource should be added.